SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/StorageAdapter.cs

```diff
@@ -151,6 +152,83 @@ private List<object[]> ExecuteSelectionSqlQuery(string query, IList<object> args
             }
         }
 
+        private long GetMaxPageCount()
+        {
+            return GetPragmaValue("max_page_count");
+        }
+
+        private long GetPageCount()
+        {
+            return GetPragmaValue("page_count");
+        }
+
+        private long GetPageSize()
+        {
+            return GetPragmaValue("page_size");
+        }
+
+        private long GetPragmaValue(string valueName)
+        {
+            var result = ExecuteSelectionSqlQuery($"PRAGMA {valueName};");
+            var count = (long)(result.FirstOrDefault()?.FirstOrDefault() ?? 0L);
+            return count;
+        }
+
+        public bool SetMaxStorageSize(long sizeInBytes)
+        {
+            bool success;
+            if (_db == null)
```

Comment:
Why we don't use check here like we do in other places?
`var db = _db ?? throw new StorageException("The database wasn't initialized.");`

Reply:
Fixed